### PR TITLE
Config management using Consul backed with Git

### PR DESCRIPTION
**Pull request by ahmadiq**

### CODE REVIEW

File: store/README.md [hunk 1]
```diff
@@ -20,3 +20,35 @@ Actuator endpoints allow you to monitor and interact with your application.
 /actuator # Provides a hypermedia-based “discovery page” for the other endpoints. Requires Spring HATEOAS to be on the classpath.
 ```
 
+Externalized configuration
+==========================
+The application stack uses the jHipster consul config loader (wrapper for git2consul) and Spring Cloud's consul config,
```
Comment (rasheedamir) [inline store/README.md]: @ahmadiq get rid of this jHipster word!

File: config/git2consul.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+    "version": "1.0",
+    "repos" : [{
+        "name" : "config/store",
+        "url" : "https://github.com/ahmadiq/storeconfig.git",
```
Comment (rasheedamir) [inline config/git2consul.json]: @ahmadiq can we put all configs in one repo? and then have different path? rite?

Comment (rasheedamir) [reply config/git2consul.json]: also each config file shld be inside each microservice source code e.g. src/main/resources/config

File: store/src/main/docker/app.yml [hunk 3]
```diff
@@ -23,6 +24,18 @@ services:
             - 8600:8600
         command: consul agent -dev -ui -client 0.0.0.0
 
+    consul-config-loader:
+        image: jhipster/consul-config-loader:v0.2.2
```
Comment (rasheedamir) [inline store/src/main/docker/app.yml:28]: @ahmadiq plz create a new docker image for stakater; we always have our own docker images now

File: store/src/main/resources/config/application.yml [hunk 4]
```diff
@@ -35,7 +35,7 @@ spring:
 
 todo:
     kafka:
-        bootstrap-servers: ${KAFKA_CLUSTER}
+        bootstrap-servers: ${KAFKA_CLUSTER:localhost}
```
Comment (rasheedamir) [inline store/src/main/resources/config/application.yml]: the default value should be localhost:9092

**Comment by rasheedamir** [comment]
@ahmadiq yaar plz think of a usecase and add it to the codebase where we have some property and we change it

File: store/src/main/docker/app.yml (hunk 3 above)
Comment (ahmadiq) [reply store/src/main/docker/app.yml:28]: ok

File: config/git2consul.json (hunk 2 above)
Comment (ahmadiq) [reply config/git2consul.json]: yes we can put configs in one repo and use different paths for each microservice.

For using separate config files for each moicroservice, might need some code modifications to git2consul. i'll check, as currently i think they only support using one file.

File: store/README.md (hunk 1 above)
Comment (ahmadiq) [reply store/README.md]: ok

File: store/src/main/resources/config/application.yml (hunk 4 above)
Comment (ahmadiq) [reply store/src/main/resources/config/application.yml]: ok

**Pull request merged**